incubator/hnc/pkg/kubectl/tree.go

```diff
@@ -34,19 +34,32 @@ var (
 var treeCmd = &cobra.Command{
 	Use:   "tree",
 	Short: "Displays the hierarchy tree rooted at the given namespace",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		nnm := args[0]
+		var nnm string
+		if len(args) != 0 {
+			nnm = args[0]
+		}
 		footnotesByMsg = map[string]int{}
 		footnotes = []string{}
-		hier := getHierarchy(nnm)
-		fmt.Println(nameAndFootnotes(hier))
-		printSubtree("", hier)
-
-		if len(footnotes) > 0 {
-			fmt.Printf("\nConditions:\n")
-			for i, n := range footnotes {
-				fmt.Printf("%d) %s\n", i+1, n)
+		nsList := make([]string, 0)
+		if nnm != "" {
+			nsList = append(nsList, nnm)
+		} else {
+			allNamespaces := getAllNamespaces()
+			nsList = append(nsList, allNamespaces...)
+		}
+		for _, each := range nsList {
```

Comment:
Nit: instead of `each`, maybe call this `nnm` ("Namespace NaMe"), especially if you get rid of the existing `nnm` variable, above. The variable `each` makes sense on this line, but `getHierarchy(each)` doesn't read as well (to me) as `getHierarchy(nnm)`.